MaxText/layers/attentions.py

```diff
@@ -1394,7 +1403,15 @@ def apply_rotary_embedding(self, inputs: Array, inputs_positions: Array, name: s
 
     rope_type = self.config.rope_type.lower()
     rope_use_scale = self.config.rope_use_scale
-    if self.config.model_name.startswith("llama3.1") or rope_type.startswith("llama3.1"):
+    if self.config.model_name.startswith("llama4") and self.name == "self_attention_vision":
```

Comment:
could you replace `self.name == "self_attention_vision"` with an explicit boolean or enum parameter in the Attention?

e.g. `is_vision_attention`?

Another option is to have VisionAttention as a subclass of Attention that overrides 

class VisionAttention(Attention):
    def apply_rotary_embedding ...

Reply:
I updated to use enum RopeType to replace rope_type to model config. So we can set rope_type at attention level instead of setting one at model level.

Reply:
I see you are OOO. Could @RissyRan review this change, also changes in Llama4 and Deepseek?

Reply:
I discussed with @RissyRan offline, I ended up adding an is_vision flag to the attention module. We may add a visionAttention module in the future (see the future item in PR description). But for now we use this way to get an early version to work.